models/persistence/SqlProxyDriver.php

```diff
@@ -59,17 +59,18 @@ function connect($id, array $params)
 
     /**
      * Proxy to query
-     * @param $statement
-     * @param $params
+     * @param string $statement
+     * @param array $params
+     * @param array $types
      *
      * @return mixed
      * @throws DBALException
      */
-    public function query($statement, $params) 
+    public function query($statement, $params, array $types = []) 
```

Comment:
I think the contract `common_persistence_sql_Driver` 
has the signature : ` public function query($statement,$params);`
and i don't think changing the contract is the best solution

Reply:
This is needed by the contact changed in the linked PR. It fixes the problems we have for example in the task queue. We need this to be able to provide field types for DBAL to perform correct value conversions for booleans in PostgreSql and for null values in Spanner.

Reply:
This is a necessary in order to integrate generis signature changed. As optional parameter, it does not break the implementation